File: house-robber/house-robber.java
class Solution {
    public int rob(int[] num) {
        if (num.length == 0){
            return 0;
        }
        else{
            int with = num[0];
            int without = 0;
            for (int i = 1; i < num.length; i ++){
                int newWith = without + num[i];
                int newWithout = Math.max(with, without);
                with = newWith;
                without = newWithout;
            }
            return Math.max(with, without);
        }
    }
}

File: 198-house-robber/house-robber.ts
function rob(nums: number[]): number {
    let n=nums.length;

    if(n<=3){
        if(n==2)
            return Math.max(nums[0], nums[1]);
        if(n==1)
            return nums[0];
        if(n==3)
            return Math.max(nums[0]+nums[2], nums[1]);
    }

    for(let i=n-3;i>=0;i--){
        let f=nums[i+2]??0;
        let s=nums[i+3]??0;
        nums[i]=nums[i]+Math.max(f, s);
    }
    return Math.max(nums[0], nums[1]);
};